src/persist/benches/writer.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Benchmarks for different persistent Write implementations.
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use criterion::{criterion_group, criterion_main, Bencher, Criterion};
+
+use persist::file::FileStream;
+use persist::mem::MemStream;
+use persist::persister::Write;
+
+fn bench_write_sync<W: Write + Clone>(
+    writer: &W,
+    data: &[((String, String), u64, isize)],
+    b: &mut Bencher,
+) {
+    b.iter_with_setup(
+        || writer.clone(),
+        |mut writer| writer.write_sync(&data).expect("failed to write data"),
+    )
+}
+
+pub fn bench_writes(c: &mut Criterion) {
+    let data = vec![
+        (("key1".to_string(), "val1".to_string()), 1, 1),
+        (("key2".to_string(), "val2".to_string()), 1, 1),
+    ];
+
+    let mem_stream = MemStream::new();
+    c.bench_function("mem_write_sync", |b| {
+        bench_write_sync(&mem_stream, &data, b)
+    });
+
+    // Create a directory that will automatically be dropped after the test finishes.
+    let temp_dir = tempfile::tempdir().expect("failed to create temp directory");
+    let file_path = temp_dir.path().join(format!(
+        "bench-{}",
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("clock went backwards")
+            .as_nanos()
+    ));
+    let file_stream = FileStream::new(file_path).expect("failed to create file stream");
+    c.bench_function("file_write_sync", |b| {
+        bench_write_sync(&file_stream, &data, b)
+    });
+}
+
+criterion_group!(benches, bench_writes);
```

Comment:
I'm not 100% sure having the group / main thing is better here or better in a single top level benches file in the benches directory. Repr does it the way I've done it and interchange does it the other way - not sure if theres a clear better way or if it matters much

Reply:
also not sure. this is fine by me and we can switch it later if we decide we were wrong